File: internal/types/lottery.go
package types

import "time"

// PresentParticipant はプレゼントルーレットの参加者情報
type PresentParticipant struct {
	UserID         string    `json:"user_id" db:"user_id"`
	Username       string    `json:"username" db:"username"`
	DisplayName    string    `json:"display_name" db:"display_name"`
	AvatarURL      string    `json:"avatar_url" db:"avatar_url"`
	RedeemedAt     time.Time `json:"redeemed_at" db:"redeemed_at"`
	IsSubscriber   bool      `json:"is_subscriber" db:"is_subscriber"`
	SubscriberTier string    `json:"subscriber_tier" db:"subscriber_tier"` // "1000", "2000", "3000"
	EntryCount     int       `json:"entry_count" db:"entry_count"`         // 購入口数（最大3口）
	AssignedColor  string    `json:"assigned_color" db:"assigned_color"`   // ルーレットの色（カラーコード）
}

// PresentLottery はプレゼントルーレットの状態
type PresentLottery struct {
	IsRunning    bool                  `json:"is_running"`
	IsLocked     bool                  `json:"is_locked"`
	Participants []PresentParticipant  `json:"participants"`
	Winner       *PresentParticipant   `json:"winner,omitempty"`
	StartedAt    *time.Time            `json:"started_at,omitempty"`
}

File: crates/overlay-db/src/tests/lottery.rs
use super::test_db;
use crate::lottery;

#[test]
fn test_lottery() {
    let db = test_db();
    let p = lottery::LotteryParticipant {
        user_id: "u1".into(),
        username: "bob".into(),
        display_name: "Bob".into(),
        avatar_url: String::new(),
        redeemed_at: "2024-01-01".into(),
        is_subscriber: false,
        subscribed_months: 0,
        subscriber_tier: String::new(),
        entry_count: 1,
        assigned_color: "#ff0000".into(),
    };
    db.add_lottery_participant(&p).unwrap();

    let all = db.get_all_lottery_participants().unwrap();
    assert_eq!(all.len(), 1);
    assert_eq!(all[0].display_name, "Bob");

    db.clear_all_lottery_participants().unwrap();
    assert!(db.get_all_lottery_participants().unwrap().is_empty());
}
